webapp/channels/src/components/advanced_text_editor/use_groups.tsx

```diff
@@ -0,0 +1,99 @@
+        return isLDAPEnabled && haveIChannelPermission(state, channel.team_id, channel.id, Permissions.USE_GROUP_MENTIONS);
+    });
+
+    const useCustomGroupMentions = useSelector((state: GlobalState) => {
+        const channel = getChannel(state, channelId);
+        return isCustomGroupsEnabled(state) && haveIChannelPermission(state, channel.team_id, channel.id, Permissions.USE_GROUP_MENTIONS);
+    });
+
+    const groupsWithAllowReference = useSelector((state: GlobalState) => {
+        const channel = getChannel(state, channelId);
+        return useLDAPGroupMentions || useCustomGroupMentions ? getAssociatedGroupsForReferenceByMention(state, channel.team_id, channel.id) : null;
+    });
+
+    const channelMemberCountsByGroup = useSelector((state: GlobalState) => selectChannelMemberCountsByGroup(state, channelId));
+
+    const getGroupMentions = useCallback((message: string) => {
+        let memberNotifyCount = 0;
+        let channelTimezoneCount = 0;
+        let mentions: string[] = [];
+        if (useLDAPGroupMentions || useCustomGroupMentions) {
+            const mentionGroups = groupsMentionedInText(message, groupsWithAllowReference);
+            if (mentionGroups.length > 0) {
+                mentionGroups.
+                    forEach((group) => {
+                        if (group.source === GroupSource.Ldap && !useLDAPGroupMentions) {
+                            return;
+                        }
+                        if (group.source === GroupSource.Custom && !useCustomGroupMentions) {
+                            return;
+                        }
+                        const mappedValue = channelMemberCountsByGroup[group.id];
+                        if (mappedValue && mappedValue.channel_member_count > Constants.NOTIFY_ALL_MEMBERS && mappedValue.channel_member_count > memberNotifyCount) {
+                            memberNotifyCount = mappedValue.channel_member_count;
+                            channelTimezoneCount = mappedValue.channel_member_timezones_count;
+                        }
+                        mentions.push(`@${group.name}`);
+                    });
+                mentions = [...new Set(mentions)];
+            }
+        }
+        return {mentions, memberNotifyCount, channelTimezoneCount};
+    }, [channelMemberCountsByGroup, groupsWithAllowReference, useCustomGroupMentions, useLDAPGroupMentions]);
+
+    // Get channel member counts by group on channel switch
+    useEffect(() => {
+        if (useLDAPGroupMentions || useCustomGroupMentions) {
+            const mentions = mentionsMinusSpecialMentionsInText(message);
+
+            if (mentions.length === 1) {
+                dispatch(searchAssociatedGroupsForReference(mentions[0], teamId, channelId));
+            } else if (mentions.length > 1) {
+                dispatch(getChannelMemberCountsByGroup(channelId));
+            }
+        }
+    }, [channelId]);
```

Comment:
I have to admit I didn't think it more deep, but you are right. Something feels off. The good thing is that after this PR, everything is a bit more localized, and it will be easier to look into, and fix it if needed. :P